src/examples/dropdowns/combobox/Default.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, useEffect } from 'react';
+import debounce from 'lodash.debounce';
+import { Item, Menu, Label, Field, Dropdown, Combobox } from '@zendeskgarden/react-dropdowns';
+import { Row, Col } from '@zendeskgarden/react-grid';
+
+const options = [
+  'Asparagus',
+  'Brussel sprouts',
+  'Cauliflower',
+  'Garlic',
+  'Jerusalem artichoke',
+  'Kale',
+  'Lettuce',
+  'Onion',
+  'Mushroom',
+  'Potato',
+  'Radish',
+  'Spinach',
+  'Tomato',
+  'Yam',
+  'Zucchini'
+];
+
+const Example = () => {
+  const [selectedItem, setSelectedItem] = useState('');
+  const [inputValue, setInputValue] = useState('');
+  const [matchingOptions, setMatchingOptions] = useState(options);
+
+  /**
+   * Debounce filtering
+   */
+  const filterMatchingOptionsRef = useRef(
+    debounce((value: string) => {
+      const matchedOptions = options.filter(
+        option => option.trim().toLowerCase().indexOf(value.trim().toLowerCase()) !== -1
+      );
+
+      setMatchingOptions(matchedOptions);
+    }, 300)
+  );
+
+  useEffect(() => {
+    filterMatchingOptionsRef.current(inputValue);
```

Comment:
Would clarify using a Ref for storing a function versus using a `useCallback` style function?

Reply:
Good call, a `useCallback` would work perfectly with the `debounce` invocation directly, however the linter gave me some trouble, so I opted with a `useMemo` instead here.